### ISSUE TITLE
Contribution is incredibly hard to follow

### ISSUE THREAD

**Issue by zfc9d3f** [dnote/dnote#222]
* project/version (e.g. CLI v0.8.0): 
* operating system: Ubuntu 18.04

## Description
Never worked with the go language, and I found the go development environment incredibly hard to deal with. 
I would like to see how the front end is built, and possibly contribute - and it seems like i have to run 
```
./server/api/scripts/test-local.sh
```
but i cannot find this anywhere 

How do i get a dev server running?

**Comment by zfc9d3f** [comment]
Go environment is incredibly hard to set up for a first time user
I did `sudo apt-get install golang-go` which gave me an older version of go even after `sudo apt update`
then i uninstalled it and went with snap which gave me 1.12
i then had to install dep, which i did from `sudo apt install go-dep` which gave me `dep: WARNING: Unknown field in manifest: prune` issues which i then realized `dep version` was still working with go1.8, then i downloaded dep from source which did not set up path variables correctly so i had to manually do that. and now i'm running into GOPATH errors which I dont even feel like working with. lol

Sorry but this really turned me off from selecting go as my next language, but if there is a simpler way to just run a local server, and if i can just take a look at the client code that would be great.

**Comment by sungwoncho** [comment]
@zfc9d3f Thanks for the feedback, and I understand your frustration. I will include how to set up Golang and install `dep` in the contributor's guide.

You are right in feeling frustrated because the contributor guide is almost non-existent for the server side code as well as the JS app. I have only open sourced them two months ago and haven't had a chance to include an official guide.

However, I am currently working on a self hosting feature and a lot of work is going into making it easier for users to run Dnote locally. I will put together something in the coming week and let you know. Thanks for your interest in contributing.

**Comment by zfc9d3f** [comment]
Hope you get it worked out. Thanks.

**Comment by sungwoncho** [comment]
Could you let me know if https://github.com/dnote/dnote/blob/master/CONTRIBUTING.md is enough for you to get the development environment going? Let me know if anything is missing.
